crates/rethnet_eth/src/remote/filter.rs

```diff
@@ -0,0 +1,120 @@
+use crate::{
+    remote::{BlockSpec, ZeroXPrefixedBytes},
+    Address, B256, U256,
+};
+
+/// for specifying the inputs to eth_newFilter
+#[derive(Clone, Default, Debug, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct FilterOptions {
```

Comment:
The _address_-as-array and _topics_-as-`B256` changes are very clear, thanks for pointing those out. I fixed those in https://github.com/NomicFoundation/hardhat/pull/4197/commits/9d2dd763847da08538d004ada19684237d7481d6

The _block_ one, however, is less clear. I see that `ethers_core` supports the single block variant, but `eth_newFilter` does not, per those API docs you linked to and also per [the JSON-RPC docs](https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_newfilter) as well as [the existing hardhat implementation](https://github.com/NomicFoundation/hardhat/blob/3e4f4b36d4df88214f897d71feb37f1b51b8ee94/packages/hardhat-core/src/internal/hardhat-network/provider/filter.ts#L14-L19).

Do you think we should support the single-block variant like `ethers_core` does? Even though that's the only one of these 4 sources that indicates it?

(I forget about those API docs you linked to, because last I heard they were still a work in progress. I see the most recent tag from their repo is still marked beta. Are you sure it's finalized enough to use as a reference?)

Reply:
You're correct! The JSON-RPC API for `eth_newFilter` indeed only accepts a range. I'm mixing up two constructs. Apologies!

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/4197/commits/ec3d09830be02b2ff856edb21e1b2bb3d190499b